docs/pages/admin-guides/access-controls/sso/keycloak.mdx

```diff
@@ -0,0 +1,355 @@
+administrators to define policies like:
+
+- Only members of the "DBA" group can connect to PostgreSQL databases.
+- Developers must never SSH into production servers.
+
+The following steps configure an example SAML authentication connector matching
+Keycloak groups. You can choose to configure other options.
+
+## Prerequisites
+
+Before you get started, you’ll need:
+
+
+- An administrative account for the Admin Console to manage realms 
+  and perform administrative tasks in the keycloak admin console
+- To be in a realm other than the master realm.
+- To register one or more users in the realm directory.
+- To create at least two groups in realm directory and assign one or more
+  users to each group.
+- A Teleport role with access to maintaining `saml` resources. This is available
+  in the default `editor` role.
+
+(!docs/pages/includes/commercial-prereqs-tabs.mdx!)
+
+- (!docs/pages/includes/tctl.mdx!)
+
+## Step 1/3. Configure Keycloak
+
+### Create a SAML Client
+1. Select the Realm to be used for your SAML integration, Click **Clients** in the menu.
+
+1. Click **Create client** 
+
+   ![Create client](../../../../img/sso/keycloak/create_client.png)
+
+1. Set Client type to **SAML** and Set a Client ID to the URL for your Teleport Proxy URL
+
+   ```code
+   https://<Var name="mytenant.teleport.sh"
+   ```
+   ![SAML](../../../../img/sso/keycloak/saml.png)
+
+1. Set the **Valid Redirect URIs** field  to the URL for your Teleport Proxy Service host and Save
+
+   ```code
+   https://<Var name="mytenant.teleport.sh" />/v1/webapi/saml/acs/keycloak
+   ```
+   ![Redirect URI](../../../../img/sso/keycloak/redirect.png)
+
+   Click **Save** before proceeding to the next step.
+
+### Configure SAML Assertion Mappings
+
+1. Navigate to **Clients** in left sidebar
+
+1. Select your Teleport SAML client, Navigate to the **Client Scope** tab
+
+1. On the Client Scopes Tabs , click on the Teleport client scope
+
+   ![Edit client scope](../../../../img/sso/keycloak/client_scope.png)
+
+1. On the Mappers tab, Click **Configure a new mapper**.
+   
+   ![Configure mapper](../../../../img/sso/keycloak/configure_mapper.png)
+
+1. Select **Groups list** from the list of attribute mappings
+
+   ![Groups list](../../../../img/sso/keycloak/groups_list.png)
+
+1. Fill out the values as shown in the image below and **Save**
+
+   ![Group mapper](../../../../img/sso/keycloak/group_mapping.png)
+
+1. Repeat steps 1 - 6 to add the **User Attribute** mapper
+
+   ![Username atrribute mapper](../../../../img/sso/keycloak/attribute_mapping.png)
+
+1. Fill out the values as shown in the image below and **Save**
+
+   ![Edit client scope](../../../../img/sso/keycloak/attribute_mapping.png)
+
+   This enables using the username inside Teleport roles as the <nobr>`{{external.username}}`</nobr> property.
+
+## Step 2/3. Create a SAML connector
+
+Now, create a SAML connector resource using `tctl`.
+
```

Comment:
Keycloak includes an explicit leading `/` in the group name. This should read `--attributes-to-roles groups,/devops,dev`. It would also be appropriate to call more attention to the `/`, or how to find out what your group names actually are in Keycloak. A screenshot of the keycloak users/groups screen could also be handy, but I don't think our other SSO guides go to that level of detail.